### PR TITLE
Filter resources based on platforms

### PR DESCRIPTION
**Pull request by barthy1**
# Changes

Hub resources have information about platforms they can run on. It makes sense to let user to filter the resources based on the platforms. This patch provides this option based on platforms information on resource level and
- creates store for platforms
- injects platforms into resource store
- adds PlatformFilter for the left pane
- extends updateUrl logic with platform parameter

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [x] Run UI Unit Tests, Lint Checks with `make ui-check`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/PlatformFilter/PlatformFilter.test.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { FakeHub } from '../../api/testutil';
+import { createProviderAndStore } from '../../store/root';
+import { when } from 'mobx';
+import { Checkbox } from '@patternfly/react-core';
+import PlatformFilter from '.';
```
Comment (vinamra28) [inline ui/src/containers/PlatformFilter/PlatformFilter.test.tsx:7]: nit:
can you please organise all the imports in the way that all external dependencies are together at top and internal together?

**Comment by PuneetPunamiya** [comment]
Thanks for the contribution 🙂 
/approve
/meow

**Pull request merged**